block/produce.go

```diff
@@ -120,6 +120,12 @@ func (m *Manager) produceApplyGossip(ctx context.Context, allowEmpty bool, nextP
 		return nil, nil, fmt.Errorf("gossip block: %w", err)
 	}
 
+	m.State.LastGossipedHeight = block.Header.Height
```

Comment:
why updating it here and not in the `gossipBlock` function? that way seems much less error-prone to have one place (cause now for example in the node recovery function for gossiping were not updating it)

Reply:
moved inside function